Update global.json references to latest available versions

benhopkinstech opened the pull request:
- Updated `xcopy-msbuild` in `global.json` to the latest available version on dotnet-eng
- Updated the dotnet 9 sdk references to the latest available version (these were out of sync)
- Updated the dotnet 8 runtime reference to the latest available version

I am raising this as I made the change to update xcopy-msbuild in [aspnetcore](https://github.com/dotnet/aspnetcore) via https://github.com/dotnet/aspnetcore/pull/60666 and spotted this repository was using the previous version also

benhopkinstech commented:
@dtivel / @erdembayar / @joelverhagen

Could one of you provide a review on this?

dtivel commented:
Thanks, @benhopkinstech.  Does this change also require a bump in the .NET SDK version?  aspnetcore is using 10.*, while this repo uses [9.0.102](https://github.com/dotnet/sign/blob/main/global.json#L4C20-L4C27).  This repo should update to 9.0.200, but I want to understand if the 2 changes need to be done together.

benhopkinstech commented:
@dtivel I have included the following in this change also

- Updated the dotnet 9 sdk references to the latest available version (these were out of sync)
- Updated the dotnet 8 runtime reference to the latest available version

dotnet 10.* should not be required for this afaik
[fsharp global.json](https://github.com/dotnet/fsharp/blob/main/global.json) is using sdk 9.0.200 and I have merged an update for xcopy-msbuild there
